### PR TITLE
Making links in task logs as hyperlinks by preventing HTML injection

### PR DESCRIPTION
**Pull request by amoghrajesh**
related: #35280 

We had to revert the change https://github.com/apache/airflow/pull/36826 due to the potential risk of HTML injection. To fix that properly, I have added a fix which does the following:
1. If any tags are present in the log lines, it will sanitize it by escaping the HTML so that the UI wont render it
2. If not, it will proceed as usual, which is convert it to a hyperlink

Added the following in a task of a DAG to test it:
```
        print("http://example.com")
        print("http://apache.org/</a><b>bad-stuff!</b>")
        print("back to normal")
        print("<b>does this work?</b>")
        print("<div onmouseover=\"alert('hi');\">hello</div>")
```

Results look like so:
image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by amoghrajesh** [comment]
@jedcunningham @bbovenzi tried fixing it properly this time, let me know what you think.

@jedcunningham: I used the cases that you had used to test, many thanks :D

File: airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts [hunk 1]
```diff
@@ -97,6 +101,17 @@ export const parseLogs = (
         line.includes(fileSourceFilter)
       )
     ) {
+      const tagRegex = /<[a-z][\s\S]*>/i;
```
Comment (bbovenzi) [inline airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts]: Perhaps we should use a real library to sanitize our string: https://www.npmjs.com/package/sanitize-html

Comment (amoghrajesh) [reply airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts]: Yeah makes sense, working on it

**Comment by amoghrajesh** [comment]
@bbovenzi attempted it and for the same log messages as the description, these are the results:
image


It sanitizes any harmful HTML tags and only allows anchor tags converts the lines with links to hyperlinks

**Comment by amoghrajesh** [comment]
@bbovenzi attempted it and for the same log messages as the description, these are the results:
image


It sanitizes any harmful HTML tags and only allows anchor tags converts the lines with links to hyperlinks

**Comment by amoghrajesh** [comment]
@bbovenzi pushed the changes using the sanitize-html library, can you take a look?

File: airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts [hunk 2]
```diff
@@ -21,6 +21,8 @@
 
 import { defaultFormatWithTZ } from "src/datetime_utils";
 
+const sanitizeHtml = require("sanitize-html");
```
Comment (bbovenzi) [inline airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts]:
```suggestion
import sanitizeHtml from "sanitize-html";
```

File: airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts [hunk 3]
```diff
@@ -97,7 +99,30 @@ export const parseLogs = (
...
     ) {
-      parsedLines.push(parsedLine);
+      // sanitize the lines to remove any tags that may cause HTML injection
+      const sanitizedLine = sanitizeHtml(parsedLine, {
+        allowedTags: ["a"],
+        allowedAttributes: {
+          a: ["href", "target", "style"],
+        },
+        transformTags: {
+          a: (tagName: any, attribs: { style: string }) => {
```
Comment (bbovenzi) [inline airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts]: Let's do the types correctly with `yarn add -D @types/sanitize-html`

Then I think it should automatically infer the types for tagName and attribs

File: airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts (hunk 2 above)
Comment (amoghrajesh) [reply airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts]: Cool, made the changes

File: airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts (hunk 3 above)
Comment (amoghrajesh) [reply airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts]: You are correct, made the change and it did take the types type implicitly

**Review by bbovenzi — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**